ai/src/exec.py

```diff
@@ -8,3 +10,11 @@ def somewhat_safe_exec(code: str, locals_dict: dict) -> dict:
 
 def somewhat_safe_eval(code: str, locals_dict: dict) -> any:
     return eval(code, {"__builtins__": {}}, locals_dict)  # pylint: disable=eval-used
+
+
+def load_module_from_code(code: str, module_name: str):
+    spec = importlib.util.spec_from_loader(module_name, loader=None)
+    module = importlib.util.module_from_spec(spec)
+
+    exec(code, module.__dict__)  # pylint: disable=exec-used
```

Comment:
Using `module_spec.loader.exec_module(module)` seems better : this will not trigger unsafe exec warning from linter and won't run anything meant for manual run (with the `if __name__ == "__main__":` test).

Reply:
Hmm, I'm not specifying a loader here since there isn't a loader for loading from a string. I guess it would be possible to write `code` to a file and then use `spec_from_file_location`, but that seems non-optimal. Do you know of a way to get a loader that loads from a string?

Reply:
After seeing `testutils-Snapshotter-SnapshotMultiJson-trader-external-reward.yaml.json` I see that you feed the code as a string in the manifest.

I was expecting the name of the file to load so people can code a python file directly, then we can use `importlib.util.spec_from_file_location` that will have a default loader.

Reply:
Ideally, we'd have all of the filesystem interaction happen in the Go runtime. This allows us to keep the I/O interface for the Python to be only gRPC/HTTP APIs. The only filesystem interaction the Python project has right now is for saving and loading the keras models - which isn't ideal, but I wasn't able to find a way to keep the keras models loaded in memory and be able to inference on them in a multi-threaded way.